lib/signaling/v2/cancelableroomsignalingpromise.js

```diff
@@ -12,106 +12,123 @@ const {
 
 const { flatMap } = require('../../util');
 
-function createCancelableRoomSignalingPromise(token, wsServer, localParticipant, iceServerSource, encodingParameters, preferredCodecs, options) {
+function createCancelableRoomSignalingPromise(token, wsServer, localParticipant, encodingParameters, preferredCodecs, options) {
   options = Object.assign({
     PeerConnectionManager: DefaultPeerConnectionManager,
     RoomV2: DefaultRoomV2,
     Transport: DefaultTransport
   }, options);
 
   let transport;
+  const log = options.log;
 
   const PeerConnectionManager = options.PeerConnectionManager;
   const RoomV2 = options.RoomV2;
 
-  const peerConnectionManager = new PeerConnectionManager(iceServerSource, encodingParameters, preferredCodecs, options);
+  const peerConnectionManager = new PeerConnectionManager(encodingParameters, preferredCodecs, options);
 
   const trackSenders = flatMap(localParticipant.tracks, trackV2 => [trackV2.trackTransceiver]);
 
-  peerConnectionManager.setConfiguration(options);
-  peerConnectionManager.setTrackSenders(trackSenders);
-
-  const cancelationError = new Error('Canceled');
+  const cancellationError = new Error('Canceled');
 
   return new CancelablePromise(function onCreate(resolve, reject, isCanceled) {
-    peerConnectionManager.createAndOffer().then(function createAndOfferSucceeded() {
-      // NOTE(mmalavalli): PeerConnectionManager#createAndOffer() queues the
-      // initial offer in the event queue for the 'description' event. So,
-      // we are dequeueing to prevent the spurious 'update' message sent by
-      // the client after connecting to a room.
-      peerConnectionManager.dequeue('description');
-
-      return new Promise((resolve, reject) => {
-        if (isCanceled()) {
-          reject(cancelationError);
-          return;
-        }
+    return new Promise((resolve, reject) => {
```

Comment:
@makarandp0 I don't think this Promise is necessary anymore. Can't we do something like:
```js
return new CancelablePromise((resolve, reject, isCanceled) => {
  if (isCanceled()) {
    reject(cancellationError);
    return;
  }
  transport = new Transport(...);
  const ensureIceServers = options.iceServers ? Promise.resolve(iceServers) : waitFor(transport, 'iced');
  ensureIceServers.then(iceServers => {
    if (isCanceled()) {
      reject(cancellationError);
      return;
    }
    ...
    return peerConnectionManager.createAndOffer();
  }).then(() => {
    if (isCanceled()) {
      reject(cancellationError);
      return;
    }
    return waitFor(transport, 'connected');
  }).then(initialState => {
    if (isCanceled()) {
      reject(cancellationError);
      return;
    }
    transport.once('stateChanged', () => {...});
    resolve(new RoomV2(...));
  });
});
```

Reply:
Thanks @manjeshbhargav for the suggestion - I have refactored file file now to make it simpler.